spec-tests/src/test/java/org/candlepin/spec/content/OwnerContentSpecTest.java

```diff
@@ -1157,4 +957,1008 @@ public void shouldPermitSafeParallelModification() throws Exception {
+                activeOwnerKeys.addAll(existing.activeOwners());
+            }
+
+            ProductInfo pinfo = new ProductInfo(product, owner, activeOwnerKeys);
+            this.productMap.put(product.getId(), pinfo);
+
+        }
+
+        private void commitGlobalSubscriptions(Collection<SubscriptionDTO> subscriptions) throws Exception {
+            Map<String, List<SubscriptionDTO>> subscriptionMap = new HashMap<>();
+
+            for (SubscriptionDTO subscription : subscriptions) {
+                NestedOwnerDTO owner = subscription.getOwner();
+
+                subscriptionMap.computeIfAbsent(owner.getKey(), key -> new ArrayList<>())
+                    .add(subscription);
+            }
+
+            for (Map.Entry<String, List<SubscriptionDTO>> entry : subscriptionMap.entrySet()) {
+                String ownerKey = entry.getKey();
+                List<SubscriptionDTO> ownerSubs = entry.getValue();
+
+                AsyncJobStatusDTO job;
+                if (CandlepinMode.isStandalone()) {
+                    File manifest = new ExportGenerator()
+                        .addSubscriptions(ownerSubs)
+                        .export();
+
+                    job = this.adminClient.owners()
+                        .importManifestAsync(ownerKey, List.of(), manifest);
+                }
+                else {
+                    ownerSubs.forEach(subscription -> this.adminClient.hosted()
+                        .createSubscription(subscription, true));
+
+                    job = this.adminClient.owners()
+                        .refreshPools(ownerKey, false);
+                }
+
+                assertThatJob(job)
+                    .isNotNull()
+                    .terminates(this.adminClient)
+                    .isFinished();
+            }
+        }
+
+        @BeforeAll
+        public void setup() throws Exception {
+            // Ensure we have our required test extensions or we'll be very broken...
+            this.checkRequiredExtensions();
+
+            this.adminClient = ApiClients.admin();
+
+            this.owners = List.of(
+                this.createOwner("owner1"),
+                this.createOwner("owner2"),
+                this.createOwner("owner3"));
+
+            this.productMap = new HashMap<>();
+            this.contentMap = new HashMap<>();
+
+            List<SubscriptionDTO> subscriptions = new ArrayList<>();
+
+            // Dummy org we use for creating a bunch of future pools for our global products. We'll also
+            // create per-org subscriptions for these products later. Also note that these will never be
+            // fully resolved.
+            OwnerDTO globalOwner = this.createOwner("global");
+
+            List<ProductDTO> globalProducts = new ArrayList<>();
+            for (int i = 1; i <= 3; ++i) {
+                ContentDTO content = Contents.random();
+
+                ProductContentDTO productContent = new ProductContentDTO();
```

Comment:
You can use the fluent-style mutators to chain these:
```
ProductContentDTO productContent = new ProductContentDTO()
  .content(content)
  .enabled(true);
```

Reply:
Updated